### PR TITLE
doc: add support for development release pages

### PR DESCRIPTION
**Pull request by utzig**
Add support for development release versions in versions.json. Any release that uses a `-xyz` suffix is considered a development release and gets a distinct banner. The first release declared in the "VERSIONS" key in versions.json that uses only numbering is considered the last release.

### CODE REVIEW

File: doc/static/js/ncs.js [hunk 1]
```diff
@@ -101,15 +118,21 @@ function NCS () {
...
+
     const OLD_RELEASE_MSG =
-      "You are looking at an older version of the documentation. You might " +
-      "want to switch to the documentation for the <a href='" +
-      last_release_url + "'>" + VERSIONS[1] + "</a> release or the " +
-      "<a href='" + latest_release_url + "'>current state of development</a>.";
+      "You are looking at an older version of the documentation. " + SWITCH_MSG;
+
+    const DEV_RELEASE_MSG =
+      "You are looking at a development version of the documentation. " + SWITCH_MSG;
```
Comment (ru-fu) [inline doc/static/js/ncs.js:135]:
```suggestion
      "You are looking at the documentation for a development release. " + SWITCH_MSG;
```

File: doc/static/js/ncs.js [hunk 2]
```diff
@@ -101,15 +118,21 @@ function NCS () {
...
 
     const LAST_RELEASE_MSG =
       "You are looking at the documentation for the last official release.";
```
Comment (ru-fu) [inline doc/static/js/ncs.js:138]:
```suggestion
      "You are looking at the documentation for the latest official release.";
```
Unrelated to this PR, but we got feedback that "last" is misleading.

**Review by ru-fu — COMMENTED** [review]
I tested it and it seems to work - but we will need to replace the ncs.js file (resp. other files that have the content) for all previous releases so that the message contains the correct number for the latest release. Correct?

**Comment by utzig** [comment]
> I tested it and it seems to work - but we will need to replace the ncs.js file (resp. other files that have the content) for all previous releases so that the message contains the correct number for the latest release. Correct?

Yes.

File: doc/static/js/ncs.js (hunk 1 above)
Comment (utzig) [reply doc/static/js/ncs.js]: Done.

File: doc/static/js/ncs.js (hunk 2 above)
Comment (ru-fu) [reply doc/static/js/ncs.js:138]: Could you change it here, or should I do a separate PR? ;)

**Review by ru-fu — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**